app/models/host/base.rb

```diff
@@ -175,7 +173,8 @@ def set_interfaces(parser)
       end
 
       parser.interfaces.each do |name, attributes|
-        iface = get_interface_scope(name, attributes).first || interface_class(name).new(:managed => false)
+        iface_scope = get_interface_scope(name, attributes)
```

Comment:
Thanks, that clears it up. The `nil` return value was introduced in 7b75a6a, and it looks to me like the fallback for creating a new interface will be correct.